adapters/adhese/adhese.go

```
@@ -0,0 +1,247 @@
+
+	host, err := macros.ResolveMacros(*&a.endpointTemplate, endpointParams)
+	if err != nil {
+		errs = append(errs, WrapError("Could not compose url from template and request account val: "+err.Error()))
+		return nil, errs
+	}
+	complete_url := fmt.Sprintf("%s%s%s%s%s",
+		host,
+		extractSlotParameter(params),
+		extractTargetParameters(params),
+		extractGdprParameter(request),
+		extractRefererParameter(request))
+
+	return []*adapters.RequestData{{
+		Method: "GET",
+		Uri:    complete_url,
+	}}, errs
+}
+
+func (a *AdheseAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{WrapError(fmt.Sprintf("Unexpected status code: %d.", response.StatusCode))}
+	}
+
+	var bidResponse openrtb.BidResponse
+
+	var adheseBidResponseArray []AdheseBid
+	if err := json.Unmarshal(response.Body, &adheseBidResponseArray); err != nil {
+		return nil, []error{err, WrapError(fmt.Sprintf("Response %v could not be parsed as generic Adhese bid.", string(response.Body)))}
+	}
+
+	var adheseBid = adheseBidResponseArray[0]
+
+	if adheseBid.Origin == "JERLICIA" {
+		var extArray []AdheseExt
+		var originDataArray []AdheseOriginData
+		if err := json.Unmarshal(response.Body, &extArray); err != nil {
+			return nil, []error{err, WrapError(fmt.Sprintf("Response %v could not be parsed to JERLICIA ext.", string(response.Body)))}
+		}
+
+		if err := json.Unmarshal(response.Body, &originDataArray); err != nil {
+			return nil, []error{err, WrapError(fmt.Sprintf("Response %v could not be parsed to JERLICIA origin data.", string(response.Body)))}
+		}
+		bidResponse = convertAdheseBid(adheseBid, extArray[0], originDataArray[0])
+	} else {
+		bidResponse = convertAdheseOpenRtbBid(adheseBid)
+	}
+
+	price, _ := strconv.ParseFloat(adheseBid.Extension.Prebid.Cpm.Amount, 64)
```

Please do not ignore the errors from `ParseFloat` and `ParseUint`.